### PR TITLE
Gh-2988: Add tinkerpop process standard suite

### LINKED ISSUE #2988: Look into adding core tinkerpop tests
**Describe the new feature you'd like**
As described in https://github.com/gchq/Gaffer/issues/277#issuecomment-254504980, we should maybe hook the `GafferPopGraph` up to `StructureStandardSuite` and `ProcessStandardSuite` so the standard tests can be run on it. This may flag up issues with our TinkerPop implementation which need fixing. For example, we might claim to implement some [features](https://tinkerpop.apache.org/docs/current/reference/#_features) that we don't.

### PR DESCRIPTION
**Pull request by l46978**
Initial integration of the standard TinkerPop process test suite.

Current results are:
`Tests run: 229, Failures: 69, Errors: 83, Skipped: 9`

Further work to be done to improve the broken tests.

# Related issue

- Resolve #2988

### CODE REVIEW

File: library/tinkerpop/src/test/resources/gaffer/map-store.properties
Comment (GCHQDeveloper314) [inline library/tinkerpop/src/test/resources/gaffer/map-store.properties:1]: What was the reason for changing this file? Removing the cache services prevents named operation tests from working. This needs to be reverted to fix those tests.

File: library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java [hunk 1]
```diff
@@ -311,6 +330,19 @@ public void addVertex(final GafferPopVertex vertex) {
...
+                .build())
+        .build();
+
+        final Iterable<? extends GafferPopElement> result = execute(new Builder()
+                .first(getOperation)
+                .then(new GenerateObjects.Builder<GafferPopElement>()
+                        .generator(new GafferPopElementGenerator(this))
+                        .build())
+                .build());
+
```
Comment (GCHQDeveloper314) [inline library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java:345]: Are these used? It doesn't look like it based on the other change. I reckon the annotations are the only changes needed in this file.

**Review by GCHQDeveloper314 — CHANGES_REQUESTED** [review]
Tests are failing due to the `map-store.properties` file change.

File: library/tinkerpop/src/test/resources/gaffer/map-store.properties
Comment (GCHQDeveloper314) [reply library/tinkerpop/src/test/resources/gaffer/map-store.properties:1]: If these changes are required for the tinkerpop suite, then hopefully that can use a different file instead.

File: library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java (hunk 1 above)
Comment (l46978) [reply library/tinkerpop/src/main/java/uk/gov/gchq/gaffer/tinkerpop/GafferPopGraph.java:345]: Must have missed this, will remove. Thank you

**Review by GCHQDeveloper314 — APPROVED** [review]

**Pull request merged**